app/lib/collection-role.coffee

```diff
@@ -0,0 +1,12 @@
+apiClient = require '../api/client'
+
+module.exports =
+  hasSettingsRole: ->
+    if @props.user?
+      apiClient.type('collection_roles').get(collection_id: @props.collection.id, user_id: @props.user.id)
+        .catch ->
+          false
+        .then ([userRoles]) ->
+          userRoles.roles.some (role) -> role.match ///owner|collaborator///i
```

Comment:
sorry, did you mean something like this before i check for the roles.. `return false if userRoles.length == 0`